### ISSUE TITLE
Combining picking and replenishment stations into one?

### ISSUE THREAD

**Issue by imgeorgiev** [merschformann/RAWSim-O#10]
I was wondering if it is possible to combine both the picking and replenishment stations into one?

If not, I would be happy to implement support for that but I would appreciate some guidance :)

Would it be straightforward to do or are all the control algorithms assuming independent stations and would require heavy re-work to enable them to work?

**Comment by merschformann** [comment]
Yeah, I discussed this question with colleagues during my research before. I never looked at it in more detail. My excuse was that surrounding warehouse processes may already be one-directional (conveyors, etc.). However, I think it is an interesting question and certainly something an RMFS can support easily (from the electromechanical perspective :) ).

The easiest thing that comes to my mind for RAWSim-O would be to simply write a layout generator that supports stations with a replenishment and order picking point. Basically, an Input-Station right next to an Output-Station.
The problem would be that the two waypoints for the handover cannot be the same (for path planning / reservation system reasons).

The other option would be to implement something like an `InputOutputStation` that fulfills both roles. I would have to look into this in more detail too (it's been a while 😅 ), but I think your intuition is correct. The biggest problem will be the different control algorithms that expect the stations to be two separate entities. Thinking about it, it may not be too dramatic though / the mechanisms may still function. Order Assignment can still assign orders until the station is at capacity. Same goes for replenishment assignment. Something like BalancedTaskAllocation will send "twice" the robots to the station as it thinks it's two stations, but that may just be a configuration issue. In the end, it may require some different control mechanisms / algorithms to get good performance, but investigating that is what RAWSim-O was designed for anyway. 😹 
It may also get a bit tricky to change the layout generators. It's hopefully sufficient to just have another case in addition to Input- and Output-Stations. 🤔

**Comment by imgeorgiev** [comment]
Thanks for the help and for giving me some direction! I'll look into it and hopefully make a PR :)
